client/ayon_deadline/addon.py

```diff
@@ -81,6 +91,157 @@ def get_server_info_by_name(self, server_name: str) -> DeadlineServerInfo:
 
         return server_info
 
+    def submit_job(
+        self,
+        server_name: str,
+        plugin_info: Dict[str, Any],
+        job_info: Dict[str, Any],
+        aux_files: Optional[List[str]] = None,
+    ) -> Dict[str, Any]:
+        """Submit job to Deadline.
+
+        Args:
+            server_name (str): Deadline Server name from project Settings.
+            plugin_info (dict): Plugin info data.
+            job_info (dict): Job info data.
+            aux_files (Optional[List[str]]): List of auxiliary files.
+
+        Returns:
+            str: Job ID.
+
+        """
+        payload = {
+            "JobInfo": job_info,
+            "PluginInfo": plugin_info,
+            "AuxFiles": aux_files or [],
+        }
+        server_url, auth, verify = self._get_deadline_con_info(server_name)
+        response = requests.post(
+            f"{server_url}/api/jobs",
+            json=payload,
+            timeout=10,
+            auth=auth,
+            verify=verify
+        )
+        if not response.ok:
+            raise ValueError("Failed to create job")
+
+        payload["job_id"] = response.json()["_id"]
+        return payload
+
+    def submit_ayon_plugin_job(
+        self,
+        server_name: str,
+        args: "Union[List[str], str]",
+        job_name: str,
+        batch_name: str,
+        job_type: Optional[JobType] = None,
+        department: Optional[str] = None,
+        chunk_size: Optional[int] = 1,
+        priority: Optional[int] = 50,
+        initial_status: Optional["InitialStatus"] = "Active",
+        group: Optional[str] = None,
+        pool: Optional[str] = None,
+        secondary_pool: Optional[str] = None,
+        output_directories: Optional[List[str]] = None,
+        output_filenames: Optional[List[str]] = None,
+        username: Optional[str] = None,
+        comment: Optional[str] = None,
+        env: Optional[Dict[str, str]] = None,
+        dependency_job_ids: Optional[List[str]] = None,
+        custom_job_info: Optional[Dict[str, Any]] = None,
+        aux_files: Optional[List[str]] = None,
+        frames: Optional[str] = None,
+    ) -> Dict[str, Any]:
```

Comment:
At first glance, this needs a docstring.

Secondly, isn't there a reason there's a global `JobInfo` (data)class in the `ayon_deadline` repo so that we can pass it along, instead of needing to create functions and methods with this argument list.

I really figured this should've just been `job_info: JobInfo` as argument. Which greatly simplifies the implementation?

Reply:
> I really figured this should've just been job_info: JobInfo as argument. Which greatly simplifies the implementation?

I actually needed this to pass in dictionary. We might support both if you think it's worth it. It should be as easy as `job_info.serialize()`.

But the issue is that most of the job related classes and functions in `lib.py` should be moved to some `pipeline.py` which is used only in hosts (plugins) because the file requires imports from core pipeline, which in ideal case scenario should not really happen in addon import.

Reply:
I think we should support passing in an explicit `JobInfo` and have that be the entry point for the addon to do anything with it. But maybe I'm overthinking it. It feels odd that we're somehow duplicating that data class with this explicit method with all these separate arguments.

It'd also mean as soon as we expose more job info attributes that are relevant we need to update the signature of this method instead of just expanding the `JobInfo` data class?

Reply:
I see your point, but the way how the `AYONDeadlineJobInfo` is being used is not intuitive. CollectJobInfo does create `AYONDeadlineJobInfo` which is then used as source for render jobs too...

Reply:
Changed, as I found out `AYONDeadlineJobInfo` is not meant as AYON plugin job info, I did small refactor of the code.